staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -584,6 +586,8 @@ func GetValidationDirective(cmd *cobra.Command) (string, error) {
 	b, err := strconv.ParseBool(validateFlag)
 	if err != nil {
 		switch validateFlag {
+		case "":
```

Comment:
Do you even need these cases?
```
		case "":
			return metav1.FieldValidationStrict, nil
		case cmd.Flag("validate").NoOptDefVal:
			return metav1.FieldValidationStrict, nil
```

It looks like this would be enough now that `strict` is the default.
```
		case "strict":
			return metav1.FieldValidationStrict, nil
```

Just trying to simplify things a little.

Reply:
These checks were here because of this test https://github.com/kubernetes/kubernetes/blob/5b274249400157028f1d9107b6bbd0686be2c076/staging/src/k8s.io/kubectl/pkg/cmd/util/helpers_test.go#L480

Because this https://github.com/kubernetes/kubernetes/blob/5b274249400157028f1d9107b6bbd0686be2c076/staging/src/k8s.io/kubectl/pkg/cmd/util/helpers_test.go#L521 behaves like `--validate=`(not like `--validate`) and causing test failure.

But to simplify things, I removed conditions and instead fixed test.